public/components/trace_analytics/components/dashboard/dashboard_content.tsx

```diff
@@ -75,59 +84,120 @@ export function DashboardContent(props: DashboardProps) {
   useEffect(() => {
     let newFilteredService = '';
     for (const filter of filters) {
-      if (filter.field === 'serviceName') {
-        newFilteredService = filter.value;
-        break;
+      if (mode === 'data_prepper') { 
+        if (filter.field === 'serviceName') {
+          newFilteredService = filter.value;
+          break;
+        }
+      } else if (mode === 'jaeger') {
+        if (filter.field === 'process.serviceName') {
+          newFilteredService = filter.value;
+          break;
+        }
       }
     }
     setFilteredService(newFilteredService);
-    if (!redirect && indicesExist) refresh(newFilteredService);
-  }, [filters, startTime, endTime, appConfigs]);
+    if (!redirect && ((mode === 'data_prepper' && dataPrepperIndicesExist) || (mode === 'jaeger' && jaegerIndicesExist))) refresh(newFilteredService);
+  }, [filters, startTime, endTime, redirect, mode, dataPrepperIndicesExist, jaegerIndicesExist]);
```

Comment:
appConfigs is coming from app analytics rendering this component but causes infinite retry with timeout of jaeger query

Reply:
So not sure if we can remove it

Reply:
What does appConfigs of App analytics actually contain/will they never change since each app is set and cannot be changed?

Reply:
Initial testing with dashboard in app analytics seems like it will be ok - this page will be re-rendered on any app config change

Reply:
its app filters

https://github.com/opensearch-project/dashboards-observability/blob/3dd96c5f7d18ea27c10745603b74a790dd9ae43e/public/components/trace_analytics/components/common/filters/filters.tsx#L36

Reply:
what triggered the infinite retry?

Reply:
adding a comment here for future reference. appConfigs is always an empty array for non-application dashboard, when toast changes and dashboard content re-renders, the array is reallocated and triggers useEffect. setting to a dummy useState array fixes it.

need to (in the future) find out if the empty array is being used, and remove it for non-app dashboard if not used